react/week-3/learning-objectives.md

```diff
@@ -1,22 +1,23 @@
 ## Learning Objectives - React Week 3
 
-### Lifecycle
+### Data fetching (advanced)
 
-- "Clean up" a component when it is unmounted
-
-### Data fetching
-
-- Show some data from an API in a component
-  - Understand that `render` will be called before data is available and handle this appropriately
-  - Use `componentDidMount` lifecycle method to trigger `fetch()`
-  - Be able to hold data, loading & error states and render correctly based on these
+- Can explain what the effect of unmounting a component
+- Cleaning up effects
+  - Recognise when an effect could cause a memory leak
+  - Be able to prevent an effect causing a memory leak using the unsubscribe function of `useEffect()`
```

Comment:
What example of a clean-up do you have in mind?

Reply:
I have been playing around with updating the "clock" example that was used in the old content:

- [Starting point](https://codesandbox.io/s/toggleable-clock-with-hooks-starting-point-gtqif?file=/src/Clock.js)
- [Mid point](https://codesandbox.io/s/toggleable-clock-with-hooks-mid-point-xqjn1?file=/src/Clock.js)
- [Final version](https://codesandbox.io/s/toggleable-clock-with-hooks-final-9ir6w?file=/src/Clock.js)

The idea being that the clock in the mid-point has a memory leak - check the console, when `Time` is unmounted - that they would have to fix by adding an unsubscribe function.

I'm vaguely aware that timers and `useEffect` can be a bit awkward, but I think for this example it works ok. And I think demonstrating a (minor) memory leak is somewhat useful too.

Reply:
I've written it up as part of my (upcoming) PR, it's a bit hard to tell how much time it will take to work through. My sense is that it won't take too long, but we can revisit this discussion when I open my PR?